src/persist-client/src/write.rs

```diff
@@ -132,41 +132,176 @@ where
     {
         trace!("WriteHandle::append lower={:?} upper={:?}", lower, upper);
 
-        let upper = upper;
-        let since = Antichain::from_elem(T::minimum());
-        let mut desc = Description::new(lower.clone(), upper, since);
+        let iter = updates.into_iter();
```

Comment:
let's add something like a `batch_from_iter` (maybe even could just be named `batch`?) method to WriteHandle that both of these can use